subsys/net/l2/wifi/wifi_shell.c

```diff
@@ -58,6 +59,23 @@ static struct net_mgmt_event_callback wifi_shell_mgmt_cb;
 		}							\
 	} while (false)
 
+#define parse_number(sh, param, str, min, max)		\
```

Comment:
I think this should be converted to a helper function to prevent the warning, I don't really see another way around it. It shouldn't really bloat the actual code if multiple parses were grouped within a single `if` check, I was thinking of something like:
```
if (!parse_number(...) ||
    ...
    !parse_number(...)) {
	return -EINVAL;
}

```